arch/xtensa/src/common/xtensa_coproc.S

```diff
@@ -112,77 +113,75 @@ _xtensa_coproc_savestate:
 
 	/* Move the address of the thread state save area to R15 */
 
-	mov		a15, a2							/* A15 is now the address of the save area */
-
+	mov	a3, a2							/* A15 is now the address of the save area */
+	addi	a3, a3, (4 * XCPTCONTEXT_REGS)
```

Comment:
Why `4 * XCPTCONTEXT_REGS` and not `XCPTCONTEXT_SIZE`?  `XCPTCONTEXT_SIZE` is the full interrupt frame and it contains extra bytes required by the ABI, is that taken under consideration when saving/restoring?